docs/src/pages/[platform]/components/collection/demo.tsx

```diff
@@ -11,46 +10,52 @@ import {
   Heading,
   Collection,
   FlexContainerStyleProps,
+  useTheme,
 } from '@aws-amplify/ui-react';
 
-import { Example } from '@/components/Example';
-import { GetFieldControls } from '../shared/GetFieldControls';
-import { useFlexContainerStyleProps } from '../shared/useFlexContainerStyleProps';
+import { Demo } from '@/components/Demo';
+import { CollectionPropControls } from './CollectionPropControls';
+import { useCollectionProps } from './useCollectionProps';
+import {
+  filterDemoProps,
+  getDemoProps,
+  objectEntriesToPropString,
+} from '../utils/demoProps';
+
+const propsToCode = (props) => {
+  const filteredProps = filterDemoProps(props);
+  return `
+<Collection
+  type='list'
+${objectEntriesToPropString(Object.entries(filteredProps))}
+>
+  {collectionItems}
+</Collection>`;
+};
 
 export const CollectionDemo = () => {
-  const flexStyleProps = useFlexContainerStyleProps({
-    alignItems: 'normal',
-    alignContent: 'start',
+  const defaultValues = {
     direction: 'row',
     gap: '20px',
-    justifyContent: 'space-between',
     wrap: 'nowrap',
-  });
-  const PropControls = GetFieldControls({
-    typeName: 'Collection List (Flex)',
-    fields: flexStyleProps,
-  });
-  const [
-    [alignItems],
-    [alignContent],
-    [direction],
-    [gap],
-    [justifyContent],
-    [wrap],
-  ] = flexStyleProps;
+  };
+
```

Comment:
Can you use [`demoState`](https://github.com/aws-amplify/amplify-ui/blob/main/docs/src/utils/demoState.ts#L20) to preserve the demo state so that it will not get wiped out when switching between light and dark mode? Example <- https://github.com/aws-amplify/amplify-ui/blob/main/docs/src/pages/%5Bplatform%5D/components/loader/demo.tsx#L26-L28